### ISSUE TITLE
App Crashes on startup.

### ISSUE THREAD

**Issue by Zigkurat** [msintuneappsdk/ms-intune-app-sdk-ios#212]
**Describe the bug:**
Application crashed on startup during internal testing. The test device was not enrolled. Can't figure out the case, because Intune SDK is obfuscated.

**To Reproduce**
There are no clear steps to reproduce, crash happened only once.

**Expected behavior:**
We expect that app never crashes on startup.

**Crash Log:**
```
Fatal Exception: NSInvalidArgumentException
Application tried to present modally a view controller <CMARiFVsoGYgVzeEPXqFrxXl: 0x131e28330> that is already being presented by <CMARiJhVaRorSvINWAYnrKKM: 0x131e27f40>.

0  CoreFoundation                 0x9905c __exceptionPreprocess
1  libobjc.A.dylib                0x15f54 objc_exception_throw
2  UIKitCore                      0x4d637c -[UIViewController _presentViewController:withAnimationController:completion:]
3  UIKitCore                      0x43d3d8 __63-[UIViewController _presentViewController:animated:completion:]_block_invoke_2
4  UIKitCore                      0x174064 +[UIView(Animation) performWithoutAnimation:]
5  UIKitCore                      0x2c0a34 __63-[UIViewController _presentViewController:animated:completion:]_block_invoke
6  UIKitCore                      0x25cdfc -[UIViewController _performCoordinatedPresentOrDismiss:animated:]
7  UIKitCore                      0x3d3638 -[UIViewController _presentViewController:animated:completion:]
8  UIKitCore                      0x2e5de0 -[UIViewController presentViewController:animated:completion:]
9  IntuneMAMSwift                 0x20d63c (Missing)
10 IntuneMAMSwift                 0x20e324 (Missing)
11 IntuneMAMSwift                 0x215770 (Missing)
12 IntuneMAMSwift                 0x15ef98 (Missing)
13 IntuneMAMSwift                 0x1a40ac (Missing)
14 libdispatch.dylib              0x1c04 _dispatch_call_block_and_release
15 libdispatch.dylib              0x3950 _dispatch_client_callout
16 libdispatch.dylib              0x11d30 _dispatch_main_queue_callback_4CF
17 CoreFoundation                 0x51ce4 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__
18 CoreFoundation                 0xbebc __CFRunLoopRun
19 CoreFoundation                 0x1f3c8 CFRunLoopRunSpecific
20 GraphicsServices               0x138c GSEventRunModal
21 UIKitCore                      0x51b0bc -[UIApplication _run]
22 UIKitCore                      0x298be8 UIApplicationMain
23 IntuneMAMSwift                 0x20a27c (Missing)
24 MyApplication                     0x31ddc8 main + 21 (main.m:21)
25 ???                            0x107421a24 (Missing)
```

**SDK Initialization code:**
This function is called from `application: didFinishLaunchingWithOptions:`

```
- (void)setupWithTennantId:(NSString *)tennantId
                     appId:(NSString *)appId
               redirectUrl:(NSString *)redirectUrl {
    IntuneMAMSettings.aadAuthorityUriOverride = tennantId;
    IntuneMAMSettings.aadClientIdOverride = appId;
    IntuneMAMSettings.aadRedirectUriOverride = redirectUrl;
    IntuneMAMEnrollmentManager *intuneMAMEnrollmentManager = [IntuneMAMEnrollmentManager instance];
    intuneMAMEnrollmentManager.delegate = self;
    if (intuneMAMEnrollmentManager.enrolledAccount == nil) {
        self.retryMamLogin = YES;
        [intuneMAMEnrollmentManager loginAndEnrollAccount:nil];
    } else {
        [self createInitialViewController];
    }
}
```

The delegate method looks like this:

```
- (void)enrollmentRequestWithStatus:(IntuneMAMEnrollmentStatus *)status {
    dispatch_async(dispatch_get_main_queue(), ^{
        if (status.didSucceed) {
            [self createInitialViewController];
        } else if (self.retryMamLogin) {
            self.retryMamLogin = NO;
            [IntuneMAMEnrollmentManager.instance loginAndEnrollAccount:nil];
        } else {
            exit(0);
        }
    });
}
```

**Smartphone (please complete the following information):**
 - Device: iPhone XR
 - OS: iOS 15.0.0

**Intune App SDK for iOS (please complete the following information):**
- What version of the Intune SDK are you using? Are you using the latest version?
14.7.0
- What platform is your app based in (native, Xamarin based, Cordova, etc)?
native
- For errors post build, does the app launch without being Intune SDK integrated?
yes

**Comment by MohamedHusseinIOS** [comment]
same issue

**Comment by OmarHassan25** [comment]
Same issue here

**Comment by abasore** [comment]
@Zigkurat are you experiencing this issue on the newest SDK version? 15.2.0?

**Comment by abasore** [comment]
It may be easier for us to provide support for this issue through a support ticket. By filing a support ticket, we could look at log and crash log files. If you have access to crash log files, you can file a support ticket by going to Endpoint Manager (https://endpoint.microsoft.com -> "Troubleshooting + support" -> "Help and support" -> "Intune").

**Comment by Zigkurat** [comment]
@abasore as I mentioned, It happened only once, so I can't tell if it's present in the newest SDK version or not. 
The crash log is very clear, and I don't think it will be hard to find what's causing the crash if you have the source code on your hands :)
This ticket was opened more as a "Just So You Know".

**Comment by abasore** [comment]
@Zigkurat @OmarHassan25 and @MohamedHusseinIOS we have a fix for this issue that will be available in the next release (15.3.0). Once that is available, I'll go ahead and close this issue.

**Comment by Zigkurat** [comment]
@abasore Great to hear! Thanks for your responsiveness!
